src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -313,7 +313,6 @@ private Connection newConnection() throws Exception
 
          connection = (username == null) ? dataSource.getConnection() : dataSource.getConnection(username, password);
          setupConnection(connection);
-         lastConnectionFailure.set(null);
```

Comment:
as name lastConnectionFailure suggests, to keep it till getAndSet() call,
nullifying it after connection creation is going to remove 'last' failure happened else where.
however, no harm in reporting 'very old' failure